enterprise/server/action_cache_server_proxy/action_cache_server_proxy_test.go

```diff
@@ -0,0 +1,112 @@
+package action_cache_server_proxy
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/action_cache_server"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/stretchr/testify/require"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+func runACServer(ctx context.Context, env *testenv.TestEnv, t *testing.T) repb.ActionCacheClient {
+	acServer, err := action_cache_server.NewActionCacheServer(env)
+	require.NoError(t, err)
+	grpcServer, runFunc := testenv.RegisterLocalGRPCServer(env)
+	repb.RegisterActionCacheServer(grpcServer, acServer)
+	go runFunc()
+	conn, err := testenv.LocalGRPCConn(ctx, env)
+	require.NoError(t, err)
+	return repb.NewActionCacheClient(conn)
+}
+
+func runACProxy(ctx context.Context, client repb.ActionCacheClient, env *testenv.TestEnv, t *testing.T) repb.ActionCacheClient {
+	env.SetActionCacheClient(client)
+	proxyServer, err := NewActionCacheServerProxy(env)
+	require.NoError(t, err)
+	grpcServer, runFunc := testenv.RegisterLocalGRPCServer(env)
+	repb.RegisterActionCacheServer(grpcServer, proxyServer)
+	go runFunc()
+	conn, err := testenv.LocalGRPCConn(ctx, env)
+	require.NoError(t, err)
+	return repb.NewActionCacheClient(conn)
```

Comment:
do these get cleaned up when the test is finished?

Reply:
whoops, they do now